### ISSUE TITLE
QueryRoutes API - amt ->amt_msat changing ?

### ISSUE THREAD

**Issue by LNBIG-COM** [lightningnetwork/lnd#3689]
I am developing a rebalancer and it seems to me that `QueryRoutes` method should have a amount value in millisatoshis not satoshis. For example:

I want to get a path for payment:

A -> B -> C -> D

Here are A & D - my nodes. May be first and last nodes are same (A === D)

If i will try to find path for this payment by `QueryRoutes` - the LND will not find because he cannot pay to hirself. Therefore rebalancers use a tricky trick: they use this API method for finding of path A -> B -> C but the C node in request gets amount + fee of hop (C->D). When results are gotten from LND they add last hop (C->D) to path. But there is big problem - quantity of amounts in satoshis. I can’t subtly set the amount for the node C so that then I can easily recalculate the fees of the last node (C) and the forwarded amount (I have to recount (and other rebalancers too - i saw sources) the entire chain of hops, since initially the LND server calculated them for the amount received by the C node of a multiple of 1000 millisatoshis (because `amt` satoshis), but I need to quantize in millisatoshis).

In addition, there is another argument for such changes: if you analyze your trends in API changes, then you move away from satoshis and move towards millisatoshis units (gRPC messages: Route, Hop - there all units in satoshis marked as depricated).

In other words the value of the results from the `QueryRoutes` method drops when satoshis units of measure are fed to it because these results cannot be easy used in rebalancing methods.

What do you think?

**Comment by joostjager** [comment]
I think we should fully support millisats on the rpc level. Fix in #3706
